### PR TITLE
refactor: unify the KvBackend

### PR DESCRIPTION
**Pull request by WenyXu**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

1. Unify the KvBackend
2. Rename kv_store to kv_backend

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

**Review by MichaelScofield — APPROVED** [review]

File: src/common/meta/src/kv_backend.rs [hunk 1]
```diff
@@ -127,3 +128,12 @@ where
         }
     }
 }
+
+pub trait ResettableKvStore: KvBackend
```
Comment (fengjiachun) [inline src/common/meta/src/kv_backend.rs:132]:
```suggestion
pub trait ResettableKvBackend: KvBackend
```

File: src/meta-srv/src/service/store/cached_kv.rs [hunk 2]
```diff
@@ -55,27 +55,27 @@ impl CheckLeader for AlwaysLeader {
 ///   4. Only the leader node can delete this metadata for the same reason mentioned above.
 pub struct LeaderCachedKvStore {
```
Comment (fengjiachun) [inline src/meta-srv/src/service/store/cached_kv.rs:56]:
```suggestion
pub struct LeaderCachedKvBackend {
```

**Comment by WenyXu** [comment]
@fengjiachun PTAL

**Review by fengjiachun — APPROVED** [review]
Looks great to me

**Pull request merged**